recipes/llvm-openmp/all/cmake/conan-llvm-openmp-vars.cmake.in

```diff
@@ -0,0 +1,199 @@
+# Reproduces the output of https://github.com/Kitware/CMake/blob/v3.30.2/Modules/FindOpenMP.cmake
+
+# The OpenMP_RUNTIME_MSVC input variable is ignored and always set to 'llvm' for the llvm-openmp package.
+
+# The _openmp_get_compiler_spec_date() and _openmp_set_version_by_spec_date() functions are adapted from CMake's FindOpenMP.cmake module,
+# which is distributed under the following BSD 3-Clause license:
+#
+# Copyright 2000-2024 Kitware, Inc. and Contributors
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# * Redistributions of source code must retain the above copyright
+#   notice, this list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright
+#   notice, this list of conditions and the following disclaimer in the
+#   documentation and/or other materials provided with the distribution.
+#
+# * Neither the name of Kitware, Inc. nor the names of Contributors
+#   may be used to endorse or promote products derived from this
+#   software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# For Conan v1 compatibility
+foreach(_suffix RELEASE DEBUG RELWITHDEBINFO MINSIZEREL)
+    if(DEFINED OpenMP_OpenMP_INCLUDE_DIRS_${_suffix})
+      set(_v1_suffix _${_suffix})
+    endif()
+endforeach()
+
+# Due to a non-global library location, the library cannot be consumed via OpenMP_C_FLAGS/OpenMP_CXX_FLAGS alone.
+# The OpenMP::OpenMP_C and OpenMP::OpenMP_CXX targets must be used instead.
+# Setting the flags for internal use only.
+set(_OpenMP_FLAGS "@OpenMP_FLAGS@")
+
+set(OpenMP_C_FLAGS "--dont-use-OpenMP_C_FLAGS--use-OpenMP::OpenMP_C-target-instead")
```

Comment:
This ugly, but passing the appropriate `-I`/`-L` paths was not possible since the `-L` flag does not appear to be propagated to the linker when passed via `CMAKE_C_FLAGS`, which is how this variable is typically used.

Fortunately, most modern CMake projects use the OpenMP targets instead and the rest can be patched quite easily. Here's a list of such projects (ticked ones have a patch available): https://gist.github.com/valgur/cbd1a856a2eb42c285a2360ecceb0405

The only project that really needed the flags variable instead of CMake targets was VTKm: https://gitlab.kitware.com/vtk/vtk-m/-/blob/master/CMake/VTKmDeviceAdapters.cmake#L64